api/api/management/commands/backfillmoderationdecision.py

```diff
@@ -92,14 +92,18 @@ def handle(self, *args, **options):
             for report, decision in zip(reports_chunk, decisions):
                 report.decision = decision
             MediaReport.objects.bulk_update(reports_chunk, ["decision"])
-            MediaDecisionThrough.objects.bulk_create(
+            throughs = MediaDecisionThrough.objects.bulk_create(
                 [
                     MediaDecisionThrough(
                         media_obj_id=report.media_obj_id, decision=decision
                     )
                     for report, decision in zip(reports_chunk, decisions)
                 ]
             )
+            # Bulk create does not call ``save()`` so the mark-sensitive/deindex
+            # actions will not be undertaken.
+            for through in throughs:
+                through.perform_action()
```

Comment:
This loop is a disappointing solution to the backfill, and somewhat defeats the purpose of the bulk creates above :confused: They're still useful as is, but this really goes against that effort by executing an individual query for each through instance regardless, especially because `perform_action` not only does the queries to create the deleted/sensitive model instances, but also relies on `self.decision.action`, which implicitly queries for `self.decision`, as well as `self.media_obj` which implicitly queries the entire media object. So the `perform_action` function is potentially very inefficient, especially when we consider it in light of the known case of bulk moderation actions, which must be able to operate on many records on an ongoing basis.

We can avoid the `self.media_obj` query within the method pretty easily. `self.decision.action` in this particular case (and potentially all cases where it would be relevant) is actually implicitly avoided (though honestly only accidentally, and is only an accident, not necessary) because the model will already have `decision` loaded from when it was created on line 98. I don't think we should rely on that implicit (accidental) fact to avoid the implicit query.

Instead, we should go ahead and make `perform_action` presume a bulk operation, and can be trivially adapted to do a single record more easily if we wanted (though I don't think it's necessary). Because we know decisions will be created in bulk, and because the bulk operation should perform the same as the non-bulk operation, I think we need to go ahead and solve that, potentially even down to the ES level. Unfortunately, that's a big task, and probably worth separating into a new PR instead of including all the fixes here.

Maybe something like this would work:

```
MediaObjectIdentifiers = namedtuple("identifier", "pk")
def decide(
    media_object_identifiers: list[MediaObjectIdentifiers],
    report_pks: list[int],
    action: str,
):
    # create decision
    # bulk update the reports
    # bulk create the through records (using `decision_id=decision.pk` rather than the full decision object)
    # bulk create the appropriate sensitive/deleted media instances based on the action
    # bulk update the filtered and non-filtered indexes using `Elasticsearch::bulk`
```

Anyway, I think either we should do that in this PR, or leave out this fix to the backfill and address it in a separate PR. I am strongly against introducing this kind of operation into the codebase, especially if it could accidentally be taken as an approved pattern to use when implementing the actual bulk decision creation.

Reply:
I've reverted the changes to the backfill command to implement this separately.